src/Nethermind/Nethermind.Consensus/Tracing/GethStyleTracer.cs

```diff
@@ -84,7 +84,7 @@ public GethLikeTxTrace Trace(Hash256 blockHash, int txIndex, GethTraceOptions op
         try
         {
             Dictionary<Address, AccountOverride>? stateOverride = options.StateOverrides;
-            using IOverridableTxProcessingScope? scope = stateOverride != null ? _env.BuildAndOverride(block.Header, stateOverride) : null;
+            using IOverridableTxProcessingScope? scope = _env.BuildAndOverride(block.Header, stateOverride);
```

Comment:
why do we need it always?

Reply:
We have to always apply `block.Header`. As for the state override, if it's null, then it's just ignored inside `BuildAndOverride`, but header should be always used, otherwise we just calculated it above without purpose